Content.Server/Body/Components/BloodstreamComponent.cs

```diff
@@ -1,28 +1,130 @@
-using Content.Server.Atmos;
 using Content.Server.Body.Systems;
-using Content.Shared.Atmos;
 using Content.Shared.Chemistry.Components;
+using Content.Shared.Damage;
+using Content.Shared.Damage.Prototypes;
 using Content.Shared.FixedPoint;
-using Robust.Shared.Analyzers;
-using Robust.Shared.GameObjects;
-using Robust.Shared.Serialization.Manager.Attributes;
-using Robust.Shared.ViewVariables;
+using Content.Shared.Sound;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
 
 namespace Content.Server.Body.Components
 {
     [RegisterComponent, Friend(typeof(BloodstreamSystem))]
-    public class BloodstreamComponent : Component
+    public sealed class BloodstreamComponent : Component
     {
+        public static string DefaultChemicalsSolutionName = "chemicals";
+        public static string DefaultBloodSolutionName = "bloodstream";
+        public static string DefaultBloodTemporarySolutionName = "bloodstreamTemporary";
+
+        public float AccumulatedFrametime = 0.0f;
+
+        /// <summary>
+        ///     How much is this entity currently bleeding?
+        ///     Higher numbers mean more blood lost every tick.
+        ///
+        ///     Goes down slowly over time, and items like bandages
+        ///     or clotting reagents can lower bleeding.
+        /// </summary>
+        /// <remarks>
+        ///     This generally corresponds to an amount of damage and can't go above 100.
+        /// </remarks>
+        [ViewVariables(VVAccess.ReadWrite)]
+        public float BleedAmount;
```

Comment:
this should probably just be a status effect, but no real reason to do that yet?